src/Main.ts

```diff
@@ -1120,6 +1120,74 @@ export class Main {
         }
     }
 
+    public async getChannelInfo(opts: {
```

Comment:
I'd drop the opts object here and have them as real parameters, it's a bit of an ugly design pattern.

Reply:
Unwrapped the parameters.